### PR TITLE
Pr103x L1T update centrality,  saturation fixes, EG eta restriction, jetphi ring subtraction (rebased)

### PR DESCRIPTION
**Pull request by rekovic**
10_3_X L1T for HeavyIon 
-Swap out chunky sandwich for jet phi ring subtraction
-Update eg eta restriction
-Update centrality classes
-Update centrality saturation handling

### CODE REVIEW

**Comment by rekovic** [comment]
please test

File: L1Trigger/L1TCalorimeter/python/caloParams_2018_v1_4_cfi.py [hunk 1]
```diff
@@ -76,16 +76,16 @@
...
+caloStage2Params.jetPUSUsePhiRing      = cms.uint32(True)
 caloStage2Params.jetBypassPUS          = cms.uint32(0)
 
 # Calibration options
 caloStage2Params.jetCalibrationType = cms.string("LUT")
 
 caloStage2Params.jetCompressPtLUTFile     = cms.FileInPath("L1Trigger/L1TCalorimeter/data/lut_pt_compress_2017v1.txt")
 caloStage2Params.jetCompressEtaLUTFile    = cms.FileInPath("L1Trigger/L1TCalorimeter/data/lut_eta_compress_2017v1.txt")
-caloStage2Params.jetCalibrationLUTFile    = cms.FileInPath("L1Trigger/L1TCalorimeter/data/lut_calib_2018v1_ECALZS_noHFJEC.txt")
+caloStage2Params.jetCalibrationLUTFile    = cms.FileInPath("L1Trigger/L1TCalorimeter/data/lut_calib_2018v1_ECALZS_noHFJEC_HI.txt")
```
Comment (fabiocos) [inline L1Trigger/L1TCalorimeter/python/caloParams_2018_v1_4_cfi.py]: @rekovic is this version specifically for HI only?

File: L1Trigger/L1TCalorimeter/src/firmware/Stage2Layer2DemuxSumsAlgoFirmwareImp1.cc [hunk 2]
```diff
@@ -144,6 +144,14 @@ void l1t::Stage2Layer2DemuxSumsAlgoFirmwareImp1::processEvent(const std::vector<
       }
   }
 
+  // calculate centrality
```
Comment (fabiocos) [inline L1Trigger/L1TCalorimeter/src/firmware/Stage2Layer2DemuxSumsAlgoFirmwareImp1.cc]: @rekovic it looks like this file revert the changes in #24752, could you please clarify?

**Comment by fabiocos** [comment]
@rekovic changes are not affecting the HI tests only, is this desired?

File: L1Trigger/L1TCalorimeter/python/caloParams_2018_v1_4_cfi.py (hunk 1 above)
Comment (rekovic) [reply L1Trigger/L1TCalorimeter/python/caloParams_2018_v1_4_cfi.py]: @fabiocos 
Short answer is yes, but this config file is not used centrally, nor in any of the workflows.  It is used for offline expert analysis.

File: L1Trigger/L1TCalorimeter/src/firmware/Stage2Layer2DemuxSumsAlgoFirmwareImp1.cc (hunk 2 above)
Comment (rekovic) [reply L1Trigger/L1TCalorimeter/src/firmware/Stage2Layer2DemuxSumsAlgoFirmwareImp1.cc]: @fabiocos 
The code is reorganized for clarity, and further fixes for saturation are implemented.
Attn @bundocka

File: L1Trigger/L1TCalorimeter/python/caloParams_2018_v1_4_cfi.py (hunk 1 above)
Comment (bundocka) [reply L1Trigger/L1TCalorimeter/python/caloParams_2018_v1_4_cfi.py]: Yes, this is the offline config that corresponds to the online settings for HI running. It won't be used anywhere unless called specifically when running the caloL2 emulator

File: L1Trigger/L1TCalorimeter/src/firmware/Stage2Layer2DemuxSumsAlgoFirmwareImp1.cc (hunk 2 above)
Comment (bundocka) [reply L1Trigger/L1TCalorimeter/src/firmware/Stage2Layer2DemuxSumsAlgoFirmwareImp1.cc]: this should now be correct, it took a few attempts to get the saturation handling in the emulator to match firmware due to some subtleties.

**Comment by rekovic** [comment]
@fabiocos 
Thanks for pointing to these plots, for example [here](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_10_3_X_2018-10-09-1100+24837/28840/136.788_RunSinglePh2017B+RunSinglePh2017B+HLTDR2_2017+RECODR2_2017reHLT_skimSinglePh_Prompt+HARVEST2017/L1T_L1TStage2CaloLayer2.html)

Again the issue with Tau DQM plots, also reported by C.Jones, we are looking into it (https://github.com/cms-sw/cmssw/issues/24279).  It is not symptomatic of this PR only, so I suggest we merge this one. And we address the instability appearing in the DQM plots separately.
Attn @thomreis @bundocka

**Comment by rekovic** [comment]
+1

**Comment by fabiocos** [comment]
+1
